docs/adr/000x-modelling-history.md

```diff
@@ -0,0 +1,343 @@
+  of a document when a new draft of a published document is created;
+- users cannot edit or remove images on a document once the first
+  edition is published;
+- the TimelineEntry model stores aspects of a document's state, resulting in it
+  needing to be queried outside a timeline context which limits flexibility
+  for the timeline.
+
+And this prevents a number of intended features for Content Publisher:
+
+- comparing different editions of a document;
+- republishing live content if there are any problems (currently a common
+  support task for Whitehall publisher);
+- showing users what changes a user made in a particular edit.
+
+## Decision
+
+This ADR proposes changes to the domain model to resolve the aforementioned
+pain points and provide a means to support the future intended features. These
+changes provide the means to store the individual editions of a document,
+each revision of the content of a document and each status an edition has held.
+As per [ADR-3](0003-initial-domain-modelling.md) it does not consider the
+option of sharing data between translations of a document as there are not
+the appropriate product decisions for this.
+
+A common theme in this modelling is the defining of
+[whether models are mutable](#approach-to-mutabilityimmutability) once
+persisted. Immutability is favoured as a means to preserve a history since
+each change requires storing a new record, implicitly creating a history.
+Where mutability is used, it is to simplify interacting with models and
+preserve Rails idioms.
+
+This ADR then considers the impact of history for [timeline](#timeline) and
+[topics](#topics), both areas where the needs for history are less
+clear. Finally, this ADR concludes with a [collated diagram](#collated-diagram)
+of the domain model concepts.
+
+### Core Concepts
+
+![Main concepts](000x/main-concepts-diagram.png)
+
+**Document**: A record that represents all versions of a piece of content in a
+particular locale. It has many editions and at any time it will have a current
+edition - shown on Content Publisher index - and potentially a live edition
+which is currently on GOV.UK. The live and current edition can be
+the same. Each iteration of a document's content is represented as a revision
+on the current edition, thus a document has many revisions. Document is a
+mutable entity that is used to store data common across all editions (such as
+first publishing date) and it is expected to be a joining point for
+document-related data that is not associated with a particular edition.
+
+**Edition**: A numbered version of a document that has been, or is
+expected to be, published on GOV.UK. It is associated with a revision
+and a status. It is mutable so that it can be a consistent object that
+joins to immutable data. It is a place where any edition-level
+database constraints can be placed, such as only one live edition can exist for
+a document. It is supported that two editions of the same document share the
```

Comment:
I don't think the scenario you've picked there is a particularly pertinent one as withdrawn doesn't make a difference. I've stayed away from referring to last revision of live edition being equal to first revision of next draft because of the change note stuff, so instead I've referenced the potential future revert aspect.